source/jormungandr/jormungandr/street_network/with_parking.py

```diff
@@ -0,0 +1,128 @@
+
+    def status(self):
+        return {'id': unicode(self.sn_system_id), 'class': self.__class__.__name__, 'modes': self.modes}
+
+    def _direct_path(
+        self, mode, pt_object_origin, pt_object_destination, fallback_extremity, request, direct_path_type
+    ):
+        response = self.street_network._direct_path(
+            mode, pt_object_origin, pt_object_destination, fallback_extremity, request, direct_path_type
+        )
+        if response and len(response.journeys):
+            self.add_parking_section_in_direct_path(response, pt_object_destination)
+
+        return response
+
+    def add_parking_section_in_direct_path(self, response, pt_object_destination):
+        logger = logging.getLogger(__name__)
+        logger.info("Creating parking section for direct path")
+
+        for journey in response.journeys:
+            section = journey.sections.add()
+            journey.nb_sections += 1
+
+            # The origin and destination of the parking section is the destination of the journey
+            section.origin.CopyFrom(pt_object_destination)
+            section.destination.CopyFrom(pt_object_destination)
+            # And we have to complete the destination of the first section ourself
+            # Because Jormun does not do it afterwards
+            journey.sections[0].destination.CopyFrom(pt_object_destination)
+
+            section.duration = self.parking_module.park_duration
+            journey.duration += self.parking_module.park_duration
+            journey.durations.total += self.parking_module.park_duration
+
+            section.type = response_pb2.PARK
+            section.id = 'section_1'
+
+            section.begin_date_time = journey.sections[0].end_date_time
+            section.end_date_time = section.begin_date_time + section.duration
+            journey.arrival_date_time += self.parking_module.park_duration
+
+    def get_street_network_routing_matrix(
+        self, origins, destinations, street_network_mode, max_duration, request, **kwargs
+    ):
+        response = self.street_network.get_street_network_routing_matrix(
+            origins, destinations, street_network_mode, max_duration, request, **kwargs
+        )
+
+        if response and len(response.rows) and len(response.rows[0].routing_response):
+            self.add_parking_time_in_routing_matrix(response.rows[0].routing_response, origins, destinations)
+
+        return response
+
+    def add_parking_time_in_routing_matrix(self, response, origins, destinations):
+        logger = logging.getLogger(__name__)
+        additionnal_parking_time = (
+            self.parking_module.park_duration
+            if len(origins) == 1
+            else self.parking_module.leave_parking_duration
+        )
+        logger.debug("Adding additionnal parking time for routing_matrix " + str(additionnal_parking_time))
+        for r in response:
+            r.duration += additionnal_parking_time
```

Comment:
As the time will depends on the coords, you'll have to call it here. Maybe it can be interesting to have a batch method on the parking module, but we can see that later.